packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp

```diff
@@ -6349,7 +6349,9 @@ namespace Tpetra {
           this->lclMatrix_, lclColMap, errStr, importLIDs, imports,
           numPacketsPerLID, constantNumPackets, distor, combineMode, atomic);
       if (!locallyCorrect) {
-        throw errStr->c_str();
+        std::ostringstream os;
+        os << "Error in unpackCrsMatrixAndCombine: " << *errStr << std::endl;
+        TEUCHOS_TEST_FOR_EXCEPTION(true, std::runtime_error, os.str());
```

Comment:
Why not something like `"Error in ...: "  + Teuchos::toString(*errStr)` or similar? I think it can avoid extra `std::ostringstream`. And I think TEUCHOS_TEST_FOR_EXCEPTION may already have `std::endl` in it.